mastering/editable_packages.rst

```diff
@@ -0,0 +1,125 @@
+
+When working in big projects with several functionalities interconnected it is recomended to avoid
+the one-and-only huge project approach in favor of several libraries each one of them specialized
+in a set of common tasks, even maintained by dedicated teams. This approach helps to isolate
+and reuse code, helps with compiling times and reduces the likelihood of including files that
+not correspond to the API of the required library.
+
+Nevertheless, in some case, it is useful to work in several libraries at the same time and see how
+the changes in one of them are propagated to the others. Following the
+:ref:`local workflow<package_dev_flow>` an user can execute the commands :command:`conan source`,
+:command:`conan install`, :command:`conan build` and :command:`conan package`, but in order to
+get the changes ready for a consumer library, it is needed the :command:`conan create` that will
+actually trigger a build to generate the binaries in the cache and will copy them and the needed
+headers to the directories specified in the ``package_info`` method of the *conanfile.py*. For each
+change in this library that the user wants to be reflected in a consumer package, Conan has to
+recreate the artifacts and copy all the stuff to the cache.
+
+What about if you just can tell Conan where to find the headers and the artifacts ready for
+consumption in your local working directory? Not need to package, just tell Conan to use those
+artifacts you have just generated with your IDE, sounds good? This is what the feature
+*editable packages* will do for you.
+
+Let's see this feature over an example where a developer is creating a ``CoolApp`` but at the same
+time he/she wants to work on ``cool/version@user/dev`` which is tightly coupled to the app.
+
+Setting it all up
+-----------------
+
+The package ``cool/version@user/dev`` is already working, the developer has the sources in a
+local folder, he/she is using whatever method to build and develop locally and can perform
+a :command:`conan create . libCool/version@user/dev` to create the package.
+
+Also, there is a *conanfile.txt* (or a more complex recipe) for the application ``CoolApp`` that
+has ``cool/version@user/dev`` among its requirements. When building this application, the
+resources of ``cool`` are used from the Conan local cache.
+
+Put a package in editable mode
+------------------------------
+
+To avoid creating the package ``cool/version@user/dev`` in the cache for every change, we are going
+to create **a link from the reference in the cache to the local working directory**:
+
+.. code-block:: bash
+
+    $ conan link <path/to/local/dev/libcool> cool/version@user/dev
+
+and all we need is a file to tell Conan what is the layout of our local project in order to provide
+the proper paths to the build system. There are several ways to achieve this objective:
+
+ * A file next to the *conanfile.py* with the name *.conan_layout*:
```

Comment:
previous sentence talks about `conanfile.txt`, but now its `conanfile.py`, which is a bit confusing. is it a different conanfile from the previously mentioned?

Reply:
The only reference to a `conanfile.txt` appears in the previous section asociated with the application `CoolApp`:

> Also, there is a conanfile.txt (or a more complex recipe) for the application CoolApp that has cool/version@user/dev among its requirements. When building this application, the resources of cool are used from the Conan local cache.

In the line you are pointing to I'm talking about the package in editable mode, `cool/version@user/dev`. Nevertheless I'm going to change the line:

> A file next to the conanfile.py with the name .conan_layout:

by

> A file next to the conanfile.py (of the package `cool`) with the name .conan_layout:

It will be less confusing this way?

Reply:
yeah, it's much less confusing now which conanfile are you referring to.